From: Takashi Iwai <tiwai@suse.de>
Date: Tue, 29 Mar 2016 18:48:07 +0200
Subject: [PATCH] ALSA: hda - Fix yet another i915 pointer leftover in error
 path
Patch-mainline: 4.4.108
References: bnc#1012382 bsc#1012767
Git-commit: 97cc2ed27e5a168cf423f67c3bc7c6cc41d12f82

commit 97cc2ed27e5a168cf423f67c3bc7c6cc41d12f82 upstream.

The hdac_acomp object in hdac_i915.c is left as assigned even after
binding with i915 actually fails, and this leads to the WARN_ON() at
the next load of the module.

Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=94736
Signed-off-by: Takashi Iwai <tiwai@suse.de>
Cc: Kouta Okamoto <kouta.okamoto@toshiba.co.jp>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 sound/hda/hdac_i915.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/hda/hdac_i915.c b/sound/hda/hdac_i915.c
index a90322a71f86..cce9ae5ec93b 100644
--- a/sound/hda/hdac_i915.c
+++ b/sound/hda/hdac_i915.c
@@ -240,6 +240,7 @@ int snd_hdac_i915_init(struct hdac_bus *bus)
 out_err:
 	kfree(acomp);
 	bus->audio_component = NULL;
+	hdac_acomp = NULL;
 	dev_info(dev, "failed to add i915 component master (%d)\n", ret);
 
 	return ret;
-- 
2.15.1

